packages/data-broker/src/services/dhis/InboundAggregateDataTranslator.js

```diff
@@ -103,7 +103,12 @@ export class InboundAggregateDataTranslator {
       this.dataSources,
       ({ dataElementCode }) => dataElementCode,
     );
-
+    // We can't always guarantee that dhis data elements can have codes (eg: Laos DHIS).
```

Comment:
Discussed: still lost, would be great to have a comment for why this is different

Reply:
It's confusing. There are 2 places that translate the analytics response from DHIS2, `DhisInputSchemeResolvingProxy,js` and `InboundAggregateDataTranslator.js`.

For example: we have a Laos DHIS2 raw analytics response without data element code (because Laos DHIS2 doesn't have codes for data elements):

```
"metaData": {
	"items": {
		"eKWzHRGV4Vy": {
			"uid": "eKWzHRGV4Vy",
			"name": "Data Element ABC",
			"dimensionItemType": "DATA_ELEMENT",
			"valueType": "NUMBER",
		}
	}
}
``` 

The translate response code in `DhisInputSchemeResolvingProxy,js` will translate the `**key**` of the items map from UID to our internal data element code in tupaia. Usually, in a metadata item object, it will also have `code: ABC`, but since Laos DHIS2 doesn't have codes, item object will not have that property

```
"metaData": {
	"items": {
		"Tupaia_Data_Element_Code_ABC": {
			"uid": "eKWzHRGV4Vy",
			"name": "Data Element ABC",
			"dimensionItemType": "DATA_ELEMENT",
			"valueType": "NUMBER",
		}
	}
}
```

And then when we reach the InboundAggregateDataTranslator.js, it will normally try to grab the code from the item object (since we iterate by Object.values(items) and we don't care about the key):
```
     const { uid, code: dhisCode, name } = item;
```

But since metadata item doesn't have any code, `dhisCode` will be undefined and throws an error here.
I guess we can grab the code from the metadata item key instead. But my only concern is the item key is not always the code (eg: if we fetch data from our regional DHIS2, it won't go through DhisInputSchemeResolvingProxy,js` pipeline, so the metadata item key will be uid in that case)

Reply:
>maybe just fill in the missing metadata item.code
Yeah that's what I was thinking too. I think the only reason it doesn't already is because of my incomplete original implementation... sorry about that.